spatial-tracking-explainer.md

```diff
@@ -0,0 +1,382 @@
+
+| Type                           | Subtype             | Example                                       |
+| -------------------------------| ------------------- | --------------------------------------------- |
+| `XRStationaryFrameOfReference` | `position-disabled` | 360 photo/video viewer                        |
+| `XRStationaryFrameOfReference` | `eye-level`         | Racing game or space simulator                |
+| `XRStationaryFrameOfReference` | `floor-level`       | Action game where you duck and dodge in place |
+| `XRBoundedFrameOfReference`    |                     | Room redecorator                              |
+| `XRUnboundedFrameOfReference`  |                     | Walking tour                                  |
+
+### XRFrameOfReference Availability
+
+**Guaranteed** The UA will always be able to provide this frame of reference 
+
+**Hardware-dependent** The UA will only be able to supply this frame of reference if running on XR hardware that supports it
+
+**Rejected** The UA will never provide this frame of reference
+
+&ast; Or polyfillable on inside-out tracking systems with anchor support.  See the section on "Floor Alignment Compatibility".
+
+| Type                           | Subtype             | Inline             | Immersive |
+| -------------------------------| ------------------- | ------------------ | ---------- |
+| `XRStationaryFrameOfReference` | `position-disabled` | Hardware-dependent | Guaranteed |
+| `XRStationaryFrameOfReference` | `eye-level`         | Hardware-dependent | Guaranteed |
+| `XRStationaryFrameOfReference` | `floor-level`       | Hardware-dependent | Guaranteed* |
+| `XRBoundedFrameOfReference`    |                     | Rejected           | Hardware-dependent* |
+| `XRUnboundedFrameOfReference`  |                     | Rejected           | Hardware-dependent |
+
+## Appendix B: Proposed partial IDL
+This is a partial IDL and is considered additive to the core IDL found in the main [explainer](explainer.md).
+
+```webidl
+//
+// Session
+//
+
+partial dictionary XRSessionCreationOptions {
+  XRFrameOfReferenceType requiredFrameOfReferenceType;
+};
+
+partial interface XRSession {
+  Promise<XRFrameOfReference> requestFrameOfReference(XRFrameOfReferenceOptions options);
+};
+
+//
+// Frames and Poses
+//
+
+partial interface XRFrame {
+  // Also listed in the main explainer.md
+  XRDevicePose? getDevicePose(optional XRFrameOfReference frameOfReference);
+  XRInputPose? getInputPose(XRInputSource inputSource, optional XRFrameOfReference frameOfReference);
+};
+
+[SecureContext, Exposed=Window]
+interface XRInputPose {
+  readonly attribute boolean emulatedPosition;
+  readonly attribute XRRay targetRay;
+  readonly attribute Float32Array? gripMatrix;
+};
+
+//
+// Coordinate System
+//
+
+[SecureContext, Exposed=Window] interface XRCoordinateSystem {
+  Float32Array? getTransformTo(XRCoordinateSystem other);
+};
+
+//
+// Frame of Reference
+//
+
+enum XRFrameOfReferenceType {
```

Comment:
As alluded to in my previous comments, I feel that this is a huge amount of complexity of enumerations, types, subtypes, and inheritance that is unwarranted by the underlying concepts they are trying to represent.

Reply:
Yeah, that's absolutely one example.  I'm not certain how common it will be, but it's worth adding to the explanation.  I'll probably put it in the section titled "Transitioning between XRFrameOfReferences"

(Another reason is actually that @toji is working on a design proposal that would make a single session capable of switching between `inline` and `immersive`.  In that situation, there will definitely need to be switching between frames of reference.)

Reply:
Added an example that'll show up in my next push